.github/workflows/e2eaiok_deltatuner_nightly_pypi.yml

```diff
@@ -2,41 +2,38 @@ name: Publish E2EAIOK Deltatuner nightly to PyPI
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - 'e2eAIOK/deltatuner/version'
 
-permissions:
-  contents: read
-  packages: write
 
 jobs:
-  e2eaiok-recdp-nightly-python-pypi:
-    runs-on: self-hosted
+  e2eaiok-dtuner-nightly-python-pypi:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
     steps:
     - uses: actions/checkout@v2
 
     - name: Set up Python
       uses: actions/setup-python@v2
 
-    - name: days since the commit date
-      run: |
-          :
-          timestamp=$(git log --no-walk --date=unix --format=%cd $GITHUB_SHA)
-          days=$(( ( $(date --utc +%s) - $timestamp ) / 86400 ))
-          if [ $days -eq 0 ]; then
-              echo COMMIT_TODAY=true >> $GITHUB_ENV
-          fi
-
     - name: Build sdist
       run: |
         pip install build wheel
         cd e2eAIOK/deltatuner
         release_version=$(cat version | head -1)
-        nightly_build_date=`date '+%Y%m%d%H'`
+        nightly_build_date=`date '+%Y%m%d%H%M'`
         nightly_version=${release_version}b${nightly_build_date}
-        echo $nightly_version > version
+        echo $nightly_version > deltatuner/version
         python3 setup.py sdist
 
     - name: Upload
-      uses: pypa/gh-action-pypi-publish@master
+      uses: pypa/gh-action-pypi-publish@release/v1
       with:
-        password: ${{ secrets.PYPI_API_TOKEN_PYRECDP }}
```

Comment:
is it publishable without token?

Reply:
@xuechendi Yes, I use the guide here: [Publishing to PyPI with a Trusted Publisher](https://docs.pypi.org/trusted-publishers/). It is a recommend way provided by PyPI and GitHub actions. I paste the introduction here:

> "Trusted publishing" is our term for using the [OpenID Connect (OIDC)](https://openid.net/connect/) standard to exchange short-lived identity tokens between a trusted third-party service and PyPI. This method can be used in automated environments and eliminates the need to use username/password combinations or manually generated API tokens to authenticate with PyPI when publishing.